File: Linear Search/linear search.c

#include<stdio.h>
int main()
{
    int i,j,n,m,f;
    int arr[100];

    printf("Enter Array size:");
    scanf("%d",&n);
    printf("Enter all number of the array: ");

    for(i=1; i<=n; i++)
    {
        scanf("%d",&arr[i]);
    }
    printf("Enter the number to search: ");
    scanf("%d",&m);

    for(i=1; i<=n; i++)
    {
        if(arr[i]==m)
        {
            f=i;
            break;
        }
    }

    if(f == -1)
        printf("number not found");
    else
        printf("%d found in position: %d",m,f);

    return 0;
}

File: Linear Search/LinearSearch.java
package algorithm.mamun;

import java.util.Scanner;

public class LinearSearch {
	
	static Scanner keyboard=new Scanner(System.in);
	
	public static void main(String[] args) {
		
		int i,j,n,m;
		int[] arr=new int[100];
		
		System.out.println("Enter Size of array: ");
		n=keyboard.nextInt();
		System.out.println("Enter all number of the array:");
		for(i=1;i<=n;i++)
		{
			arr[i]=keyboard.nextInt();
		}
		
		System.out.println("Enter the number to search: ");
		m=keyboard.nextInt();
		
		j=-1;
		for(i=0;i<=n;i++)
		{
			if(arr[i]==m)
				j=i;
		}
		
		if(j==-1)
			System.out.println(m + " not found");
		else
			System.out.println(m + " found in position: "+ j);
		
	}
}
